pkg/mapper/dynamicfile/mapper.go

```diff
@@ -46,31 +47,35 @@ func (m *DynamicFileMapper) Map(identity *token.Identity) (*config.IdentityMappi
 	}
 
 	if roleMapping, err := m.RoleMapping(key); err == nil {
-		if err := m.match(identity, roleMapping.RoleARN, roleMapping.UserId); err != nil {
+		if err := m.match(canonicalARN, roleMapping.RoleARN); err != nil {
 			return nil, err
 		}
 		return roleMapping.IdentityMapping(identity), nil
 	}
 
 	if userMapping, err := m.UserMapping(key); err == nil {
-		if err := m.match(identity, userMapping.UserARN, userMapping.UserId); err != nil {
+		if err := m.match(canonicalARN, userMapping.UserARN); err != nil {
 			return nil, err
 		}
 		return userMapping.IdentityMapping(identity), nil
-	}
 
+	}
 	return nil, errutil.ErrNotMapped
 }
 
-func (m *DynamicFileMapper) match(token *token.Identity, mappedARN, mappedUserID string) error {
+func (m *DynamicFileMapper) match(canonicalARN string, mappingARN string) error {
 	if m.userIDStrict {
 		// If ARN is provided, ARN must be validated along with UserID.  This avoids having to
 		// support IAM user name/ARN changes. Without preventing this the mapping would look
 		// invalid but still work and auditing would be difficult/impossible.
-		strippedArn, _ := arn.StripPath(mappedARN)
-		if strippedArn != "" && token.CanonicalARN != strings.ToLower(strippedArn) {
+		// compare arn additionally for IAM user if principalId is used in mapping
```

Comment:
This comment is the same as the text above, I dont think it needs to be added.

Reply:
thank you for the comments! yeah we can refactor it in future